packages/base-types/src/runtimeLogs/logs/steps/api.ts

```diff
@@ -0,0 +1,46 @@
+import { APIBodyType, APIMethod } from '@base-types/node/api';
+import { JsonArray, JsonObject, JsonValue } from 'type-fest';
+
+import { LoggingNodeType, LogLevel } from '../../utils';
+import { BaseStepLog } from '../base';
+
+interface ApiLogMessage {
+  request: {
+    method: APIMethod;
+    url: string;
+  };
+  response: {
+    statusCode: number;
+    statusMessage: string;
+  };
+}
+
+interface VerboseApiLogMessage {
+  request: {
+    headers: Record<string, string>;
+    query: Record<string, string>;
+  } & (
+    | {
+        method: Exclude<APIMethod, APIMethod.GET>;
+        bodyType: APIBodyType;
+        body: string | JsonObject | JsonArray;
+      }
+    | {
+        // GET requests can't have a body
+        method: APIMethod.GET;
+        bodyType: null;
+        body: null;
+      }
+  );
+  response: {
+    headers: Record<string, string>;
+    body: JsonValue;
```

Comment:
good callout. making this a `JsonValue` was because i tried to imitate the current way the API block is implemented in the runtime. so, only JSON. i'll update this to be a plain string. customers can access the `content-type` header in the `response.headers` object